src/views/clusteroverview/SettingsTab/ClusterTab/components/AutomaticSubscriptionRHELGuests/components/AutomaticSubscriptionForm/AutomaticSubscriptionForm.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { FC, useEffect, useState } from 'react';
+
+import { isEqualObject } from '@kubevirt-utils/components/NodeSelectorModal/utils/helpers';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import useSubscriptionData from '@kubevirt-utils/hooks/useSubscriptionData/useSubscriptionData';
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import {
+  ActionGroup,
+  Button,
+  ButtonVariant,
+  Form,
+  FormGroup,
+  Grid,
+  GridItem,
+  Skeleton,
+  TextInput,
+} from '@patternfly/react-core';
+
+import { ACTIVATION_KEYS_URL } from '../../utils/constants';
+import ActivationKeyHelpIcon from '../ActivationKeyHelpIcon/ActivationKeyHelpIcon';
+import OrganizationIDHelpIcon from '../OrganizationIDHelpIcon/OrganizationIDHelpIcon';
+
+import './AutomaticSubscriptionForm.scss';
+
+const AutomaticSubscriptionForm: FC = () => {
+  const { t } = useKubevirtTranslation();
+  const { isAdmin, loaded, loading, subscriptionData, updateSubscription } = useSubscriptionData();
+
+  const [activationKey, setActivationKey] = useState(null);
+  const [organizationID, setOrganizationID] = useState(null);
+
+  useEffect(() => {
+    if (activationKey === null && organizationID === null && loaded) {
+      setActivationKey(subscriptionData?.activationKey);
+      setOrganizationID(subscriptionData?.organizationID);
+    }
+  }, [activationKey, loaded, organizationID, subscriptionData]);
+
+  if (!loaded) return <Skeleton />;
+
+  const isDisabled =
+    !isAdmin ||
+    isEqualObject(subscriptionData, { activationKey, organizationID }) ||
+    (!isEmpty(organizationID) && isEmpty(activationKey)) ||
```

Comment:
what about a case of both are empty?
simply  || (isEmpty(organizationID)|| isEmpty(activationKey))